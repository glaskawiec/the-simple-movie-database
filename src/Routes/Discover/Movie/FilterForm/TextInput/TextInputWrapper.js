import styled from 'styled-components';

const TextInputWrapper = styled.div`
display: inline-block;
margin-right: 1.5em;
`;

export default TextInputWrapper;
