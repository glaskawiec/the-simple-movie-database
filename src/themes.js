const Themes = {
  light: {
    colors: {
      primary: '#F2355B',
      secondary: '#161616',
      background: '#FFFFFF',
      text: '#000000',
      mobileMenuOverlay: '#000000',
      card: '#FFFFFF',
      themeToggleThumb: '#161616',
      paginationHover: '#CDCDCD',
    },
  },
  dark: {
    colors: {
      primary: '#F2355B',
      secondary: '#090909',
      background: '#161616',
      text: 'hsla(0,0%,100%,0.88)',
      mobileMenuOverlay: '#FFFFFF',
      card: '#090909',
      themeToggleThumb: '#FFFFFF',
      paginationHover: '#292929',
    },
  },
};

export default Themes;
