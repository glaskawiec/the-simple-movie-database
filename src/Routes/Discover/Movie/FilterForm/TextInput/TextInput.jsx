import React from 'react';
import PropTypes from 'prop-types';
import Label from '../Select/Label';
import InputStyled from './InputStyled';
import TextInputWrapper from './TextInputWrapper';

const TextInput = ({
  label, placeholder, value, onChange,
}) => (
  <TextInputWrapper>
    <Label>{label}</Label>
    <InputStyled
      onChange={onChange}
      placeholder={placeholder}
      value={value}
    />
  </TextInputWrapper>
);

TextInput.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default TextInput;
