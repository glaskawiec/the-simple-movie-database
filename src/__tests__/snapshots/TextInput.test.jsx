import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import themes from '../../themes';
import TextInput from '../../routes/find/textInput/TextInput';

test('TextInput snapshot', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={themes.light}>
        <TextInput
          label="snap"
          onChange={jest.fn()}
          placeholder="snap"
          value="snap"
        />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
