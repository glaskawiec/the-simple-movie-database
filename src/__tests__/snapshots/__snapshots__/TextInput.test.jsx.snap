// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput snapshot 1`] = `
<div
  className="sc-htpNat fgrEkI"
>
  <label
    className="sc-bdVaJa fxuKrb"
  >
    snap
  </label>
  <input
    className="sc-bwzfXH lbjMp"
    onChange={[MockFunction]}
    placeholder="snap"
    value="snap"
  />
</div>
`;
